### PR TITLE
[Merged by Bors] - feat(cdk): add 'cdk test' command

### LINKED ISSUE #2832: Tracking:  Connector Development Kit Phase 1
CDK is a new component for Fluvio.  It facilitates the development, testing,  and deployment of connectors.   

```
$ cdk build
// build connector

$ cdk test
// build and run connector locally using current cdk project

$ cdk publish
// publish connector to hub

$ cdk deploy <connector_name>
// deploy connector from hub
```

TODO:
- [x] Build `cdk  build`
- [x] Schema validation
- [ ] Test `cdk test`
- [x] Publish
- [x] Deploy

Schema Validation:
- Load Package from TOML
- Validating Connector Config using Pkge metadta

### PR DESCRIPTION
**Pull request by galibey**
Support run connector from local source directory:
```
cdk test --config config-example.yaml
```

Closes #2832

### CODE REVIEW

File: crates/cdk/src/test.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        let cargo = Cargo::build()
+            .profile(opt.release)
+            .lib(false)
+            .package(p.package_name())
+            .extra_arguments(self.extra_arguments)
+            .build()?;
+
+        cargo.run()?;
+
+        let status = Command::new(p.target_bin_path()?)
```
Comment (sehz) [inline crates/cdk/src/test.rs:42]: this should be using deployer API so we have consistent deployment strategy

**Review by sehz — CHANGES_REQUESTED** [review]
1. Use deployer API,
2. Need CLI test

File: crates/cdk/src/test.rs (hunk 1 above)
Comment (galibey) [reply crates/cdk/src/test.rs]: Done

**Review by sehz — APPROVED** [review]
Nicely done

**Pull request closed without merging**